lms/static/scripts/frontend_apps/components/FilePickerApp.js

```diff
@@ -111,25 +120,77 @@ export default function FilePickerApp({ onSubmit }) {
    * render.
    */
   const [shouldSubmit, setShouldSubmit] = useState(false);
-  const submit = useCallback(() => setShouldSubmit(true), []);
+
+  const [deeplinkingFields, setDeeplinkingFields] = useState(
+    /** @type {Record<string,string>|null} */ (null)
+  );
+
+  const submit = useCallback(
+    /** @param {Content} content */
+    async content => {
+      async function getDeeplinkingFields() {
+        const data = {
+          ...deeplinkingAPI.data,
+          content,
+          extra_params: {
+            groupSet: groupConfig.useGroupSet ? groupConfig.groupSet : null,
+          },
+        };
+        setDeeplinkingFields(
+          await apiCall({
+            authToken: authToken,
+            path: deeplinkingAPI.path,
+            data,
+          })
+        );
+      }
+      // Set shouldSubmit to true early to get the spinner before any potential apiCall
+      setShouldSubmit(true);
+
+      // When deeplinkingAPI is present we want to call the backend to return the form
+      // fields we'll forward to the LMS to complete the DeepLinking request
+      if (content && deeplinkingAPI && !deeplinkingFields) {
+        try {
+          await getDeeplinkingFields();
+        } catch (error) {
+          setErrorInfo({
```

Comment:
What does the current error UI look like? I'm wondering if we need to reset the `shouldSubmit` state here.

Reply:
I added the reset for shouldSubmit and content. Without setting content to null the modal was left in an in-between state.

Some video demo:

https://user-images.githubusercontent.com/1433832/168266338-2fa36c8f-bcec-46ad-b97e-8fe8ad3b9166.mp4